src/main/scala/com/fortysevendeg/android/scaladays/ui/schedule/ScheduleFragment.scala

```diff
@@ -71,13 +73,25 @@ class ScheduleFragment
       (recyclerView
         <~ rvLayoutManager(layoutManager)
         <~ rvAddItemDecoration(new ScheduleItemDecorator())) ~
-        loadSchedule() ~
-        (refreshLayout <~ srlOnRefreshListener(loadSchedule(favorites = false, forceDownload = true, swipe = true))) ~
+        loadSchedule(favorites = shouldLoadFavorites()) ~
+        (refreshLayout <~ srlOnRefreshListener(loadSchedule(favorites = shouldLoadFavorites(), forceDownload = true, swipe = true))) ~
         (reloadButton <~ On.click(
-          loadSchedule(favorites = false, forceDownload = true)
+          loadSchedule(favorites = shouldLoadFavorites(), forceDownload = true)
         )))
   }
 
+  def preferences(): SharedPreferences = {
+    getActivity().getSharedPreferences(FILTER_PREFERENCES, Context.MODE_PRIVATE)
```

Comment:
I think sometimes the `getActivity` call can return null, do you mind to make this work with options?

Also, use `apply` instead of `commit` (see https://developer.android.com/reference/android/content/SharedPreferences.Editor.html#commit())

```scala
def preferences(): Option[SharedPreferences] =
    Option(getActivity()).map(_.getSharedPreferences(FILTER_PREFERENCES, Context.MODE_PRIVATE))

def shouldLoadFavorites(): Boolean =
  preferences().map(_.getBoolean(FILTER_KEY, false)).getOrElse(false)

  def saveFavorites(favorites: Boolean): Unit =
    preferences().foreach(_.edit().putBoolean(FILTER_KEY, favorites).apply())
```

Reply:
Done!